src/billing.ts

```diff
@@ -0,0 +1,166 @@
+            },
+            (res) => res.body
+        );
+    }
+
+    updatePaymentSettings(paymentMethodId: string) {
+        return this._connection.request(
+            {
+                method: "PUT",
+                path: "/_api/billing/paymentsettings",
+                headers: { tenant: this.tenantName },
+                body: { payment_method_id: paymentMethodId },
+                absolutePath: true
+            },
+            (res) => res.body
+        );
+    }
+
+    getPaymentDetailsOfPreviousMonths(limit: number) {
+        const qs: { [key: string]: any } = {};
+
+        if (limit) {
+            qs.limit = limit;
+        }
+
+        return this._connection.request(
+            {
+                method: "GET",
+                path: `/_api/billing/payments`,
+                headers: { tenant: this.tenantName },
+                qs,
+                absolutePath: true
+            },
+            (res) => res.body
+        );
+    }
+
+    getInvoices(limit: number) {
+        const qs: { [key: string]: any } = {};
+
+        if (limit) {
+            qs.limit = limit;
+        }
+
+        return this._connection.request(
+            {
+                method: "GET",
+                path: `/_api/billing/invoices`,
+                qs,
+                headers: { tenant: this.tenantName },
+                absolutePath: true
+            },
+            (res) => res.body
+        );
+    }
+
+    getCurrentInvoices() {
+        return this._connection.request(
+            {
+                method: "GET",
+                path: "/_api/billing/invoice/current",
+                headers: { tenant: this.tenantName },
+                absolutePath: true
+            },
+            (res) => res.body
+        );
+    }
+
+    getInvoiceOfSpecificMonthYear(year: number, month: number) {
+        return this._connection.request(
+            {
+                method: "GET",
+                path: `/_api/billing/invoices/${year}/${month}`,
+                headers: { tenant: this.tenantName },
+                absolutePath: true
+            },
+            (res) => res.body
+        );
+    }
+
+    getUsageOfTenant(startDate?: string, endDate?: string) {
+        const qs: { [key: string]: any } = {};
+
+        if (startDate && endDate) {
+            qs.startDate = startDate;
+            qs.endDate = endDate;
+        }
+
+        return this._connection.request(
+            {
+                method: "GET",
+                path: "/_api/billing/usage",
+                qs,
+                headers: { tenant: this.tenantName },
+                absolutePath: true
+            },
+            (res) => res.body
+        );
+    }
+
+    getUsageOfTenantForSpecificRegion(region: string, startDate?: string, endDate?: string) {
+        const qs: { [key: string]: any } = {};
+
+        if (startDate && endDate) {
+            qs.startDate = startDate;
+            qs.endDate = endDate;
```

Comment:
what is i only provide the startdate or enddate.

Reply:
updated